src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseGitDiffReportResource.java

```diff
@@ -168,47 +164,45 @@ public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForSu
      * @throws IOException     if errors occur while accessing the file system
      */
-    @GetMapping("programming-exercises/{exerciseId}/participation/{participationId}/commits/{commitHash1}/diff-report/{commitHash2}")
+    @GetMapping("programming-exercises/{exerciseId}/commits/{commitHash1}/diff-report/{commitHash2}")
     @EnforceAtLeastStudent
-    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommits(@PathVariable long exerciseId, @PathVariable long participationId,
-            @PathVariable String commitHash1, @PathVariable String commitHash2, @RequestParam(defaultValue = "USER") RepositoryType repositoryType)
-            throws GitAPIException, IOException {
-        log.debug("REST request to get a ProgrammingExerciseGitDiffReport for two commits for commit {} and commit {} of participation {}", commitHash1, commitHash2,
-                participationId);
-
-        VcsRepositoryUri repositoryUri;
-        ProgrammingExerciseParticipation participation;
-
-        if (repositoryType.equals(RepositoryType.USER)) {
-            participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
-            if (!participation.getExercise().getId().equals(exerciseId)) {
-                throw new ConflictException("A git diff report entry can only be retrieved if the participation's exercise id matches with the exercise id", ENTITY_NAME,
-                        "exerciseIdsMismatch");
-            }
-            repositoryUri = participation.getVcsRepositoryUri();
+    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommits(@PathVariable long exerciseId, @RequestParam(required = false) Long participationId,
+            @PathVariable String commitHash1, @PathVariable String commitHash2, @RequestParam(required = false) RepositoryType repositoryType) throws GitAPIException, IOException {
+        log.debug("REST request to get a diff report for two commits for commit {} and commit {} of participation {}", commitHash1, commitHash2, participationId);
+
+        VcsRepositoryUri repositoryUri = null;
+        if (participationId != null) {
+            Participation participation = participationRepository.findByIdElseThrow(participationId);
+            repositoryUri = getVcsRepositoryUri(exerciseId, repositoryType, participation);
+            participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
+        }
+        else if (repositoryType != null) {
+            ProgrammingExercise programmingExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationAndAuxiliaryRepositoriesElseThrow(exerciseId);
+            authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, programmingExercise, null);
+            repositoryUri = switch (repositoryType) {
+                case TEMPLATE -> programmingExercise.getTemplateParticipation().getVcsRepositoryUri();
```

Comment:
```suggestion
                case TEMPLATE -> programmingExercise.getTemplateRepositoryUri();
```
Is there a reason to not use this method?

Reply:
We need a `VcsRepositoryUri` object here, but `getTemplateRepositoryUri` returns a String